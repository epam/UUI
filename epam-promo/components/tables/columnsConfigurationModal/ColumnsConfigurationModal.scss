@use '../../../assets/styles/index' as *;

.main-panel {
    height: 100%;

    .group-title {
        margin-bottom: -6px;
    }
    .group-items {
        flex-direction: column;
    }
    .no-data {
        flex-direction: column;
        top: calc(50% - 35px);
        position: relative;
    }
    .h-divider {
        width: 100%;
        box-sizing: border-box;
        padding: 6px 24px;
        &::before {
            content: "";
            height: 1px;
            display: block;
            background: $gray30;
        }
    }
}

.search-area {
    padding-bottom: 6px;
}
