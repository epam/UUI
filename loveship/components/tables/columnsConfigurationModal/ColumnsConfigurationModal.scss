@use '../../../assets/styles/scss/index' as *;

.search {
    padding: 0 24px 12px 24px;
    display: flex;
    > :global(.uui-input-box) {
        flex-grow: 1;
    }
    .actions-menu {
        margin-left: 12px;
    }

    border-bottom: 1px solid $night300;
}

$footerHeight: 61;
$headerHeight: 60;
$searchHeight: 43;
$midAreaHeight1: calc(100vh - #{$footerHeight + $headerHeight + $searchHeight}px);
$midAreaHeight2: calc(80vh - #{$footerHeight + $headerHeight + $searchHeight}px);
.modal {
    @media (max-width: 640px) {
        height: 100vh;
        .container {
            height: $midAreaHeight1;
        }
    }
    @media (min-width: 641px) {
        height: 80vh;
        .container {
            height: $midAreaHeight2;
        }
    }
}

.container {
    .checkbox-container {
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        min-height: 40px;
        margin: 0 0 5px 0;

        > * {
            min-width: 210px;
        }

        > .h-divider {
            margin: 5px 24px;
            border-bottom: 1px solid $night300;
        }
    }

    .action-button {
        margin-left: 18px;
    }

    .section-title {
        font-family: $font-sans;
        margin-left: 24px;
        margin-top: 6px;
        display: flex;
        align-items: center;
        user-select: none;

        .title {
            font-weight: 600;
            font-size: 14px;
            line-height: 24px;
            padding-right: 6px;
        }
        .amount {
            padding: 0 2px;
            display: flex;
            flex-direction: column;
            flex-grow: 0;
            justify-content: center;
            align-items: center;
            background: $night300;
            border-radius: 50px;
            font-weight: 600;
            font-size: 12px;
            line-height: 18px;
            min-width: 14px;
            height: 18px;
            color: #6C6F80;
        }
    }
}

.row-wrapper {
    position: relative;
    padding: 1px 0;
    max-width: 100%;

    .row-wrapper-content {
        position: relative;
        padding-left: 2px;
        padding-right: 24px;
        border-radius: 3px;
        background-color: white;
        cursor: auto;
        user-select: none;
        display: flex;
        > .title {
            flex-grow: 1;
        }
    }

    &.not-pinned:not(:hover) {
        :global(.pin-icon-button) {
            visibility: hidden;
        }
    }

    &:hover {
        box-shadow: 0 0 1px $night300;
        &, .title, :global(.pin-icon-button), .row-wrapper-content {
            background-color: $night100;
        }
    }

    .drag-handle-wrapper {
        padding-left: 24px;
    }
    .drag-handle {
        opacity: 0.3;
        height: 24px;
    }

    &.dnd-allowed {
        .drag-handle {
            opacity: 1;
        }
        .drag-handle-wrapper {
            cursor: grab;
        }
    }

    &:global(.uui-drag-ghost) {
        @include dnd-ghost-shadow();
    }
}

.no-data {
    font-family: $font-sans;
    color: $night800;
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;

    > .text {
        font-size: 16px;
        font-weight: 600;
        line-height: 24px;
    }
    > .sub-text {
        font-size: 12px;
        line-height: 18px;
    }
}
