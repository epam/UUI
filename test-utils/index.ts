export * from './helpers';
export * from './mocks';
