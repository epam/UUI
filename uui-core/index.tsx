export * from './src/constants';
export * from './src/helpers';
export * from './src/services';
export * from './src/types';
export * from './src/data';
export * from './src/hooks';
export * from './src/i18n';
export * from './src/ssr';
