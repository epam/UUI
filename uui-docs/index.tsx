/* Required so far, as uui-build assumes .index.tsx as an entry. We'll switch to ./src/index.tsx later */
export * from './src';
