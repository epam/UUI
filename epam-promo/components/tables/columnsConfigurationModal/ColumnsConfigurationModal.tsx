import * as styles from "./ColumnsConfigurationModal.scss";
import * as React from "react";
//
import { ColumnsConfig, cx, DataColumnProps, DndActorRenderParams, IModal } from "@epam/uui-core";
import { DragHandle } from "@epam/uui-components";
import { ReactComponent as MenuIcon } from '@epam/assets/icons/common/navigation-more_vert-18.svg';
import { ReactComponent as ResetIcon } from '@epam/assets/icons/common/action-update-18.svg';
//
import { Dropdown, DropdownMenuButton, ModalBlocker, ModalFooter, ModalHeader, ModalWindow } from "../../overlays";
import { FlexRow, FlexSpacer, Panel, ScrollBars } from "../../layout";
import { Button, LinkButton } from "../../buttons";
import { Checkbox, SearchInput } from "../../inputs";
import { DropMarker } from "../../dnd";
import { i18n } from '../../../i18n';
//
import { useColumnsConfigurationState } from "@epam/uui-components";
import { PinIconButton } from "./PinIconButton";
import { useCallback } from "react";

const i18nLocal = i18n.tables.columnsConfigurationModal;
const returnByCondition = <T, F>(condition: boolean, ifTrue: T, ifFalse: F) => {
    return condition ? ifTrue : ifFalse;
};

interface IColumnsConfigView<TItem, TId, TFilter> {
    modalProps: IModal<ColumnsConfig>;
    columnsConfig: ColumnsConfig;
    defaultConfig: ColumnsConfig;
    columns: DataColumnProps<TItem, TId, TFilter>[];
}

export function ColumnsConfigurationModal<TItem, TId, TFilter>(props: IColumnsConfigView<TItem, TId, TFilter>) {
    const { modalProps, columns, columnsConfig, defaultConfig } = props;

    const renderRowContent = (
        props: { dndActorParams: DndActorRenderParams, column: DataColumnProps<TItem, TId>, isDndAllowed: boolean, isPinnedAlways: boolean },
    ) => {
        const {
            column,
            isDndAllowed,
            dndActorParams,
            isPinnedAlways,
        } = props;
        const cfg = columnsConfigLocal[column.key];
        const isSelected = cfg.isVisible;
        const isPinned = cfg.fix || isPinnedAlways;
        const wrapperClasses = cx(
            styles.rowWrapper,
            !isPinned && styles.notPinned,
            isDndAllowed && styles.dndAllowed,
            ...returnByCondition(isDndAllowed, dndActorParams.classNames, []),
        );
        const { onTouchStart, onPointerDown, ...restEventHandlers } = dndActorParams.eventHandlers;
        const wrapperAttrs = {
            className: wrapperClasses,
            ...returnByCondition(isDndAllowed, { ref: dndActorParams.ref }, {}),
            ...returnByCondition(isDndAllowed, restEventHandlers, {}),
        };
        const dragHandleAttrs = {
            className: styles.dragHandleWrapper,
            ...returnByCondition(isDndAllowed, { onTouchStart, onPointerDown }, {}),
        };

        return (
            <div { ...wrapperAttrs }>
                <div className={ styles.rowWrapperContent }>
                    <FlexRow background="white" spacing='6' cx={ styles.title }>
                        <span { ...dragHandleAttrs }><DragHandle cx={ styles.dragHandle } /></span>
                        <Checkbox
                            key={ column.key }
                            label={ column.caption }
                            value={ isSelected }
                            onValueChange={ () => toggleVisibility(column.key) }
                            isDisabled={ column.isAlwaysVisible || !!column.fix }
                        />
                    </FlexRow>
                    <PinIconButton
                        id={ column.key }
                        isPinned={ !!isPinned }
                        canUnpin={ !isPinnedAlways }
                        onTogglePin={ togglePin }
                    />
                </div>
                <DropMarker { ...dndActorParams } />
            </div>
        );
    };

    const {
        // props
        byGroup, isNoData, filterValue, columnsConfigLocal,
        // methods
        reset, apply, checkAll, togglePin, renderRows, uncheckAll, setFilterValue, toggleVisibility,
    } = useColumnsConfigurationState({ columnsConfig, columns, modalProps, defaultConfig, renderRowContent });

    const renderSectionTitle = (title: string, amount: number) => <div className={ styles.sectionTitle }>
        <span className={ styles.title }>{ title }</span>
        <span className={ styles.amount }>{ amount }</span>
    </div>;

    const renderVisible = () => {
        const amountPinned = byGroup.DISPLAYED_PINNED?.itemsFiltered?.length || 0;
        const amountUnPinned = byGroup.DISPLAYED_UNPINNED?.itemsFiltered?.length || 0;
        if (!amountPinned && !amountUnPinned) {
            return null;
        }
        const rowsPinned = renderRows(useColumnsConfigurationState.ColGroup.DISPLAYED_PINNED);
        const rowsUnpinned = renderRows(useColumnsConfigurationState.ColGroup.DISPLAYED_UNPINNED);
        return (
            <>
                { renderSectionTitle(i18nLocal.displayedInTable, amountPinned + amountUnPinned) }
                <div className={ styles.checkboxContainer }>
                    { rowsPinned }
                    { Boolean(amountPinned && amountUnPinned) && <div className={ styles.hDivider } /> }
                    { rowsUnpinned }
                </div>
            </>
        );
    };

    const renderHidden = () => {
        const amountHidden = byGroup.HIDDEN?.itemsFiltered?.length || 0;
        if (!amountHidden) {
            return null;
        }
        return (
            <>
                { renderSectionTitle(i18nLocal.hiddenInTable, amountHidden) }
                <div className={ styles.checkboxContainer }>
                    { renderRows(useColumnsConfigurationState.ColGroup.HIDDEN) }
                </div>
            </>
        );
    };
    const close = useCallback(() =>  modalProps.abort(), [modalProps]);
    return (
        <ModalBlocker blockerShadow="dark" { ...modalProps }>
            <ModalWindow cx={ styles.modal }>
                <ModalHeader title={ i18nLocal.configureColumnsTitle } onClose={ close } />
                <div className={ styles.search }>
                    <SearchInput
                        size="30"
                        value={ filterValue }
                        onValueChange={ setFilterValue }
                        placeholder={ i18nLocal.searchByColumnName }
                    />
                    <Dropdown
                        closeOnTargetClick={ true }
                        renderBody={ () =>
                            <Panel background="white" shadow={ true }>
                                <DropdownMenuButton caption={ i18nLocal.clearAllButton } onClick={ uncheckAll }/>
                                <DropdownMenuButton caption={ i18nLocal.selectAllButton } onClick={ checkAll }/>
                            </Panel>
                        }
                        renderTarget={ props =>
                            <Button
                                cx={ styles.actionsMenu }
                                { ...props }
                                fill="white"
                                icon={ MenuIcon }
                                size="30"
                                color="gray50"
                                isDropdown={ false }
                            />
                        }
                        placement="bottom-end"
                    />
                </div>
                <Panel background="white" cx={ styles.container }>
                    <ScrollBars>
                        { renderVisible() }
                        { renderHidden() }
                        {
                            isNoData &&
                            <div className={ styles.noData }>
                                <div className={ styles.text }>{ i18nLocal.noResultsFound }</div>
                                <div className={ styles.subText }>{ i18nLocal.weCantFindAnyItemMatchingYourRequest }</div>
                            </div>
                        }
                    </ScrollBars>
                </Panel>
                <ModalFooter borderTop >
                    <LinkButton icon={ ResetIcon } caption={ i18nLocal.resetToDefaultButton } color="blue" onClick={ reset } />
                    <FlexSpacer />
                    <Button fill="white" color="gray50" caption={ i18nLocal.cancelButton } onClick={ close } />
                    <Button cx={ styles.actionButton } caption={ i18nLocal.applyButton } color="green" onClick={ apply } />
                </ModalFooter>
            </ModalWindow>
        </ModalBlocker>
    );
}
