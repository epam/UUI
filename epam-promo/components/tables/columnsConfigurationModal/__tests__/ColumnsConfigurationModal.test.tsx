import React from 'react';
import { ColumnsConfigurationModal } from '../ColumnsConfigurationModal';
import { getDefaultColumnsConfig } from '@epam/uui-core';
import { Product } from '@epam/uui-docs';
import { renderWithContextAsync } from '@epam/test-utils';

const mockColumns = [
    {
        key: 'id',
        caption: 'ID',
        render: (product: Product) => <div>{ product }</div>,
        isSortable: true,
        isAlwaysVisible: true,
        grow: 0, shrink: 0, width: 96,
    },
    {
        key: 'level',
        caption: 'Level',
        render: (product: Product) => <div>{ product }</div>,
        isSortable: true,
        isAlwaysVisible: true,
        grow: 0, shrink: 0, width: 96,
    },
];

describe('ColumnsConfigurationModal', () => {
    it('should be rendered correctly', async () => {
        const modalProps = {
            isActive: true,
            key: 'test',
            zIndex: 1,
            abort: jest.fn,
            success: jest.fn,
        };
        const defaultConfig = getDefaultColumnsConfig(mockColumns);
        const tree = await renderWithContextAsync(
            <ColumnsConfigurationModal
                { ...modalProps }
                columns={ mockColumns }
                columnsConfig={ defaultConfig }
                defaultConfig={ defaultConfig }
            />);
        expect(tree).toMatchSnapshot();
    });
});


