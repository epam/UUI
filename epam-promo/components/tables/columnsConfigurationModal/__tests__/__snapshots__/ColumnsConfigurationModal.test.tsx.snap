// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnsConfigurationModal should be rendered correctly 1`] = `
<div
  className="container modalBlocker shadow-dark"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
  <div
    data-focus-lock-disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      aria-modal={true}
      className="uui-modal-window modal width-420 height-700 container"
      role="modal"
    >
      <div
        className="root size-36 background-none padding-24 vPadding-12 spacing-12 modalHeader container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="root font-sans-semibold text-color-gray80 line-height-24 font-size-18 v-padding-12 container"
        >
          Configure columns
        </div>
        <div
          className="flexSpacer"
        />
        <div
          className=""
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "auto",
              "flexGrow": undefined,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <button
            className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="root size-36 background-none padding-24 border-bottom-gray40 spacing-12 searchArea container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="container uui-input-box -clickable root size-30 mode-form uui-theme-promo"
          onBlur={[Function]}
          onFocus={[Function]}
          tabIndex={-1}
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
          <input
            className="uui-input"
            inputMode="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search by column name"
            tabIndex={0}
            type="search"
            value=""
          />
        </div>
        <button
          className="container uui-button-box uui-enabled -clickable button-vars button-gray50 root size-30 mode-outline uui-theme-promo actionsMenu"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
        </button>
      </div>
      <div
        className="root background-white mainPanel container"
      >
        <div
          className="root root theme-light"
          cx={
            Array [
              Array [
                "root",
                "theme-light",
              ],
              undefined,
            ]
          }
          style={
            Object {
              "display": "flex",
              "height": "100%",
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "bottom": 0,
                "flex": "1 1 auto",
                "left": 0,
                "marginBottom": 0,
                "marginRight": 0,
                "overflow": "scroll",
                "position": "relative",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              className="root size-24 background-none padding-24 spacing-6 groupTitle container align-items-center"
              style={
                Object {
                  "columnGap": undefined,
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="root font-sans-semibold text-color-gray80 line-height-24 font-size-14 v-padding-6 groupTitleText container"
              >
                Displayed in table
              </div>
              <button
                className="container uui-button-box uui-enabled size-18 fill-solid badge-color-gray30 root groupTitleBadge"
                onClick={[Function]}
                tabIndex={-1}
                type="button"
              >
                <div
                  className="uui-caption"
                >
                  2
                </div>
              </button>
            </div>
            <div
              className="root size-30 background-none groupItems container align-items-center"
              style={
                Object {
                  "columnGap": undefined,
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="root size-30 background-none rowWrapper notPinned -draggable container align-items-center"
                onPointerEnter={[Function]}
                onPointerLeave={[Function]}
                onPointerMove={[Function]}
                onPointerUp={[Function]}
                style={
                  Object {
                    "columnGap": undefined,
                    "rowGap": undefined,
                  }
                }
              >
                <div
                  className="root size-30 background-white spacing-6 title container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <div
                    className="dragHandle container uui-drag-handle"
                    onPointerDown={[Function]}
                    onTouchStart={[Function]}
                  />
                  <label
                    className="container uui-checkbox-container root size-18 mode-form uui-theme-promo uui-disabled"
                    tabIndex={-1}
                  >
                    <div
                      className="uui-checkbox uui-checked"
                    >
                      <input
                        aria-disabled={true}
                        checked={true}
                        disabled={true}
                        onChange={[Function]}
                        tabIndex={-1}
                        type="checkbox"
                      />
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </div>
                    <div
                      className="uui-input-label"
                    >
                      ID
                    </div>
                  </label>
                </div>
                <div
                  className="flexSpacer"
                />
                <div
                  className="root size-30 background-none pinIconButton container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <span
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <button
                      aria-disabled={false}
                      className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
                      onClick={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </button>
                  </span>
                </div>
              </div>
              <div
                className="root size-30 background-none rowWrapper notPinned -draggable container align-items-center"
                onPointerEnter={[Function]}
                onPointerLeave={[Function]}
                onPointerMove={[Function]}
                onPointerUp={[Function]}
                style={
                  Object {
                    "columnGap": undefined,
                    "rowGap": undefined,
                  }
                }
              >
                <div
                  className="root size-30 background-white spacing-6 title container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <div
                    className="dragHandle container uui-drag-handle"
                    onPointerDown={[Function]}
                    onTouchStart={[Function]}
                  />
                  <label
                    className="container uui-checkbox-container root size-18 mode-form uui-theme-promo uui-disabled"
                    tabIndex={-1}
                  >
                    <div
                      className="uui-checkbox uui-checked"
                    >
                      <input
                        aria-disabled={true}
                        checked={true}
                        disabled={true}
                        onChange={[Function]}
                        tabIndex={-1}
                        type="checkbox"
                      />
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </div>
                    <div
                      className="uui-input-label"
                    >
                      Level
                    </div>
                  </label>
                </div>
                <div
                  className="flexSpacer"
                />
                <div
                  className="root size-30 background-none pinIconButton container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <span
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <button
                      aria-disabled={false}
                      className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
                      onClick={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="uui-track-horizontal"
            style={
              Object {
                "display": "none",
                "height": 6,
                "position": "absolute",
              }
            }
          >
            <div
              className="uui-thumb-horizontal"
            />
          </div>
          <div
            className="uui-track-vertical"
            style={
              Object {
                "display": "none",
                "position": "absolute",
                "width": 6,
              }
            }
          >
            <div
              className="uui-thumb-vertical"
            />
          </div>
        </div>
      </div>
      <div
        className="root size-36 background-none padding-24 vPadding-24 spacing-12 modalFooter borderTop container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <button
          className="container uui-button-box uui-enabled -clickable root size-36 linkButton-color-blue uui-has-left-icon uui-no-right-icon"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
          <div
            className="uui-caption"
          >
            Reset to Default
          </div>
        </button>
        <div
          className="flexSpacer"
        />
        <button
          className="container uui-button-box uui-enabled -clickable button-vars button-gray50 root size-36 mode-outline uui-theme-promo"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Cancel
          </div>
        </button>
        <button
          aria-disabled={false}
          className="container uui-button-box uui-enabled -clickable button-vars button-green root size-36 mode-solid uui-theme-promo"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Apply
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
</div>
`;

exports[`ColumnsConfigurationModal should disable Apply button if all columns are hidden 1`] = `
<div
  className="container modalBlocker shadow-dark"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
  <div
    data-focus-lock-disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      aria-modal={true}
      className="uui-modal-window modal width-420 height-700 container"
      role="modal"
    >
      <div
        className="root size-36 background-none padding-24 vPadding-12 spacing-12 modalHeader container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="root font-sans-semibold text-color-gray80 line-height-24 font-size-18 v-padding-12 container"
        >
          Configure columns
        </div>
        <div
          className="flexSpacer"
        />
        <div
          className=""
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "auto",
              "flexGrow": undefined,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <button
            className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="root size-36 background-none padding-24 border-bottom-gray40 spacing-12 searchArea container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <div
          className="container uui-input-box -clickable root size-30 mode-form uui-theme-promo"
          onBlur={[Function]}
          onFocus={[Function]}
          tabIndex={-1}
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
          <input
            className="uui-input"
            inputMode="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search by column name"
            tabIndex={0}
            type="search"
            value=""
          />
        </div>
        <button
          className="container uui-button-box uui-enabled -clickable button-vars button-gray50 root size-30 mode-outline uui-theme-promo actionsMenu"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
        </button>
      </div>
      <div
        className="root background-white mainPanel container"
      >
        <div
          className="root root theme-light"
          cx={
            Array [
              Array [
                "root",
                "theme-light",
              ],
              undefined,
            ]
          }
          style={
            Object {
              "display": "flex",
              "height": "100%",
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "bottom": 0,
                "flex": "1 1 auto",
                "left": 0,
                "marginBottom": 0,
                "marginRight": 0,
                "overflow": "scroll",
                "position": "relative",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              className="root size-24 background-none padding-24 spacing-6 groupTitle container align-items-center"
              style={
                Object {
                  "columnGap": undefined,
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="root font-sans-semibold text-color-gray80 line-height-24 font-size-14 v-padding-6 groupTitleText container"
              >
                Hidden in table
              </div>
              <button
                className="container uui-button-box uui-enabled size-18 fill-solid badge-color-gray30 root groupTitleBadge"
                onClick={[Function]}
                tabIndex={-1}
                type="button"
              >
                <div
                  className="uui-caption"
                >
                  2
                </div>
              </button>
            </div>
            <div
              className="root size-30 background-none groupItems container align-items-center"
              style={
                Object {
                  "columnGap": undefined,
                  "rowGap": undefined,
                }
              }
            >
              <div
                className="root size-30 background-none rowWrapper notPinned -draggable container align-items-center"
                onPointerEnter={[Function]}
                onPointerLeave={[Function]}
                onPointerMove={[Function]}
                onPointerUp={[Function]}
                style={
                  Object {
                    "columnGap": undefined,
                    "rowGap": undefined,
                  }
                }
              >
                <div
                  className="root size-30 background-white spacing-6 title container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <div
                    className="dragHandle container uui-drag-handle"
                    onPointerDown={[Function]}
                    onTouchStart={[Function]}
                  />
                  <label
                    className="container uui-checkbox-container root size-18 mode-form uui-theme-promo -clickable"
                    tabIndex={-1}
                  >
                    <div
                      className="uui-checkbox"
                    >
                      <input
                        checked={false}
                        disabled={false}
                        onChange={[Function]}
                        tabIndex={0}
                        type="checkbox"
                      />
                    </div>
                    <div
                      className="uui-input-label"
                    >
                      ID
                    </div>
                  </label>
                </div>
                <div
                  className="flexSpacer"
                />
                <div
                  className="root size-30 background-none pinIconButton container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <span
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <button
                      aria-disabled={false}
                      className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
                      onClick={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </button>
                  </span>
                </div>
              </div>
              <div
                className="root size-30 background-none rowWrapper notPinned -draggable container align-items-center"
                onPointerEnter={[Function]}
                onPointerLeave={[Function]}
                onPointerMove={[Function]}
                onPointerUp={[Function]}
                style={
                  Object {
                    "columnGap": undefined,
                    "rowGap": undefined,
                  }
                }
              >
                <div
                  className="root size-30 background-white spacing-6 title container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <div
                    className="dragHandle container uui-drag-handle"
                    onPointerDown={[Function]}
                    onTouchStart={[Function]}
                  />
                  <label
                    className="container uui-checkbox-container root size-18 mode-form uui-theme-promo -clickable"
                    tabIndex={-1}
                  >
                    <div
                      className="uui-checkbox"
                    >
                      <input
                        checked={false}
                        disabled={false}
                        onChange={[Function]}
                        tabIndex={0}
                        type="checkbox"
                      />
                    </div>
                    <div
                      className="uui-input-label"
                    >
                      Level
                    </div>
                  </label>
                </div>
                <div
                  className="flexSpacer"
                />
                <div
                  className="root size-30 background-none pinIconButton container align-items-center"
                  style={
                    Object {
                      "columnGap": undefined,
                      "rowGap": undefined,
                    }
                  }
                >
                  <span
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <button
                      aria-disabled={false}
                      className="container uui-button-box uui-enabled -clickable icon-button-default root uui-theme-promo icon-button-color-gray60"
                      onClick={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="container uui-icon uui-enabled"
                        style={Object {}}
                      >
                        <svg
                          className=""
                        />
                      </div>
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="uui-track-horizontal"
            style={
              Object {
                "display": "none",
                "height": 6,
                "position": "absolute",
              }
            }
          >
            <div
              className="uui-thumb-horizontal"
            />
          </div>
          <div
            className="uui-track-vertical"
            style={
              Object {
                "display": "none",
                "position": "absolute",
                "width": 6,
              }
            }
          >
            <div
              className="uui-thumb-vertical"
            />
          </div>
        </div>
      </div>
      <div
        className="root size-36 background-none padding-24 vPadding-24 spacing-12 modalFooter borderTop container align-items-center"
        style={
          Object {
            "columnGap": undefined,
            "rowGap": undefined,
          }
        }
      >
        <button
          className="container uui-button-box uui-enabled -clickable root size-36 linkButton-color-blue uui-has-left-icon uui-no-right-icon"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="container uui-icon uui-enabled"
            style={Object {}}
          >
            <svg
              className=""
            />
          </div>
          <div
            className="uui-caption"
          >
            Reset to Default
          </div>
        </button>
        <div
          className="flexSpacer"
        />
        <button
          className="container uui-button-box uui-enabled -clickable button-vars button-gray50 root size-36 mode-outline uui-theme-promo"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Cancel
          </div>
        </button>
        <button
          aria-disabled={true}
          className="container uui-button-box uui-disabled -clickable button-vars button-green root size-36 mode-solid uui-theme-promo"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Apply
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
</div>
`;
